packages/lumx-react/src/utils/type.ts

```diff
@@ -91,3 +91,35 @@ export const isComponentType = (type: ReactElement['type']) => (node: ReactNode)
  * (excluding `NaN` as it can't be distinguished from `number`)
  */
 export type Falsy = false | undefined | null | 0 | '';
+
+/**
+ * Allow at least one of aria-label or arial-labelledby props to be set.
+ */
+export type WithAriaLabelOrLabelledBy =
+    | {
+          'aria-labelledby': string;
+          'aria-label'?: undefined;
+      }
+    | {
+          'aria-label': string;
+          'aria-labelledby'?: undefined;
+      };
+
+/**
+ * Require either a `label` prop to be used as aria-label by the component
+ * or one of aria-label or arial-labelledby prop.
+ */
+export type WithLabelPropOrAriaLabel<T = string | undefined> = T extends string
+    ? {
+          /**
+           * The label of the dialog.
+           * Must define the purpose of the dialog.
+           * Alternatively, you can use `aria-label` directly or target an existing element as title using `aria-labelledby`
+           */
+          label: T;
```

Comment:
It's probably best to use aria-label directly since the label is not used in DOM or elsewhere (like the IconButton)

Reply:
Got it, removed this type and kept only the other!